source/common/buffer/buffer_impl.cc

```diff
@@ -410,19 +410,21 @@ void OwnedImpl::commit(uint64_t length, absl::Span<RawSlice> slices,
   std::unique_ptr<OwnedImplReservationSlicesOwner> slices_owner(
       static_cast<OwnedImplReservationSlicesOwner*>(slices_owner_base.release()));
 
-  absl::Span<Slice> owned_slices = slices_owner->ownedSlices();
-  ASSERT(slices.size() == owned_slices.size());
+  absl::Span<Slice::SizedStorage> owned_storages = slices_owner->ownedStorages();
+  ASSERT(slices.size() == owned_storages.size());
 
   uint64_t bytes_remaining = length;
   for (uint32_t i = 0; i < slices.size() && bytes_remaining > 0; i++) {
-    Slice& owned_slice = owned_slices[i];
-    if (owned_slice.data() != nullptr) {
-      owned_slice.maybeChargeAccount(account_);
```

Comment:
Was this not needed? This seems incorrect to remove.

Reply:
Yep. This is not needed now. Because we will account the memory when we construct `Slice` from `Storage`.